### PR TITLE
boards: bbc_microbit_v2: Add buzzer

### PR DESCRIPTION
**Pull request by larsgk**
Tie pwm1 to buzzer pin and adjust sample.

### CODE REVIEW

**Comment by larsgk** [comment]
Hi @kartben - I see you were auto-assigned.  Would there be more potential reviewers than those in the list?

File: boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts [hunk 1]
```diff
@@ -113,6 +113,14 @@
 	status = "okay";
 };
 
+&pwm1 {
+	/* buzzer */
+	status = "okay";
```
Comment (kartben) [inline boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts:118]: it shouldn't be on by default and only enabled by the overlay.

**Review by kartben — CHANGES_REQUESTED** [review]
can you also update the sample.yaml?
Hopefully @anangl can help with reviewing more thoroughly and potentially assign a couple other reviewers.

Thanks!

**Comment by kartben** [comment]
@anangl could you help review?

File: boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts (hunk 1 above)
Comment (anangl) [reply boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts:118]: Unlike microbit v1, v2 has an on-board speaker, so I think it makes sense to enable the PWM for driving it at the board level.

But the documentation of the sound sample should be updated, as for microbit v2 it is not required to connect any additional hardware:
https://github.com/zephyrproject-rtos/zephyr/blob/04de43b1a119665532b6b2a0c6bbcbefd47eb7d6/samples/boards/bbc_microbit/sound/README.rst#L9-L17

Also this line from `prj.conf` should be removed (it is redundant for v1, and for v2 it generates a Kconfig warning):
https://github.com/zephyrproject-rtos/zephyr/blob/04de43b1a119665532b6b2a0c6bbcbefd47eb7d6/samples/boards/bbc_microbit/sound/prj.conf#L5

**Review by anangl — CHANGES_REQUESTED** [review]

File: boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts (hunk 1 above)
Comment (anangl) [reply boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts:118]: > So, should I leave the `status = "okay"`, @kartben @anangl ?

I'm okay with leaving it here.

**Comment by kartben** [comment]
> can you also update the sample.yaml? 

@larsgk 👆 :)

**Comment by larsgk** [comment]
> > can you also update the sample.yaml?
> 
> @larsgk 👆 :)

Sorry...missed that. Will do 🙂

**Comment by larsgk** [comment]
> > > can you also update the sample.yaml?
> > 
> > 
> > @larsgk 👆 :)
> 
> Sorry...missed that. Will do 🙂

@kartben - just added bbc_microbit_v2 to sample.yml - how do I verify it will be picked up?

**Review by kartben — DISMISSED** [review]
looks good, but you have compliance issues that'll need fixing :)
nit: might make sense to make the mentions to BBC microbit v1/v2 in the README actual links to the boards, but also OK to leave as-is.

**Review by anangl — APPROVED** [review]

**Review by kartben — APPROVED** [review]
Thanks!

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by larsgk** [comment]
> looks good, but you have compliance issues that'll need fixing :) 

I've been struggling with LE Audio all day - so I probably missed it, but what compliance issues?

**Pull request merged**